packages/framework/src/state/device.ts

```diff
@@ -0,0 +1,57 @@
+import { atom, useAtom } from "jotai";
+import { useCallback } from "react";
+
+// https://wicg.github.io/ua-client-hints/#navigatorua
+declare interface NavigatorUA {
+  readonly userAgentData?: UALowEntropyJSON;
+}
+
+// https://wicg.github.io/ua-client-hints/#dictdef-navigatoruabrandversion
+interface NavigatorUABrandVersion {
+  readonly brand: string;
+  readonly version: string;
+}
+
+// https://wicg.github.io/ua-client-hints/#dictdef-ualowentropyjson
+interface UALowEntropyJSON {
+  readonly brands: NavigatorUABrandVersion[];
+  readonly mobile: boolean;
+  readonly platform: string;
+}
+
+export interface DeviceActions {
+  setData: (values: { [key: string]: unknown }) => void;
+}
+
+export interface DeviceDefinition {
+  platform: string;
+  width: number;
+  height: number;
+  [key: string]: unknown;
+}
+
+export const defaultDevice = {
+  platform:
+    (navigator as Navigator & NavigatorUA).userAgentData?.platform || "",
```

Comment:
What's with the extra typing?

Reply:
`Navigator` type is not comes with `userAgentData`, so need to add `userAgentData` type to avoid lint errors